File: core/src/nl/t64/game/rpg/screens/inventory/ListenerKeyVertical.java
package nl.t64.game.rpg.screens.inventory;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.InputListener;
import lombok.AllArgsConstructor;

import java.util.function.IntConsumer;


@AllArgsConstructor
class ListenerKeyVertical extends InputListener {

    private final IntConsumer updateIndexFunction;

    @Override
    public boolean keyDown(InputEvent event, int keycode) {
        switch (keycode) {
            case Input.Keys.UP -> updateIndexFunction.accept(-1);
            case Input.Keys.DOWN -> updateIndexFunction.accept(1);
        }
        return true;
    }

}

File: core/src/nl/t64/game/rpg/screens/inventory/ListenerKeyVertical.kt
package nl.t64.game.rpg.screens.inventory

import com.badlogic.gdx.Input
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.InputListener


internal class ListenerKeyVertical(
    private val updateIndexFunction: (Int) -> Unit
) : InputListener() {

    override fun keyDown(event: InputEvent, keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.UP -> updateIndexFunction.invoke(-1)
            Input.Keys.DOWN -> updateIndexFunction.invoke(1)
        }
        return true
    }

}
